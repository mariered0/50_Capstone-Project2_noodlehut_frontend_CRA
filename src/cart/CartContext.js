import React from "react";

const cartContext = React.createContext();

export default cartContext;
cartContext.displayName = "Cart";