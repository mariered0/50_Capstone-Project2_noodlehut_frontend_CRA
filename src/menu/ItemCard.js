import React, { useContext } from "react";
import cartContext from "../cart//CartContext";
import {
  Grid,
  Card,
  CardContent,
  CardMedia,
  CardActions,
  Button,
  Typography,
} from "@mui/material";

import defaultImg from "../image/image_unavailable.png";


const ItemCard = ({ id, name, desc, price }) => {
  const { cart, setCart, addToCart, removeFromCart } = useContext(cartContext);

  return (
    
        <Grid item key={id} xs={12} sm={6} md={4}>
          <Card sx={{ height: '100%' }}>
            <CardMedia
              component="img"
              image={ defaultImg }
            />
            <CardContent sx={{ flexGrow: 1 }}>
              <Typography gutterBottom variant="h5" component="h2">
                {name}
              </Typography>
              <Typography>{desc}</Typography>
              <Typography>{price}</Typography>
            </CardContent>
            <CardActions>
              <Button size="small"
                      onClick={() => addToCart(id)}
              >
                Add
              </Button>
            </CardActions>
          </Card>
        </Grid>
     
  );
};

export default ItemCard;
