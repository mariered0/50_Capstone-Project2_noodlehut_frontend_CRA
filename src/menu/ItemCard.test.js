import React from "react";
import { render } from '@testing-library/react';
import ItemCard from "./ItemCard";
import CartContext from "../cart/CartContext";
const { isCartOpen, setIsCartOpen, cart, addToCart, removeFromCart } = require("../cart/_testCommon");


it("renders without crashing", () => {
    render(
        <CartContext.Provider value={{ isCartOpen, setIsCartOpen, cart, addToCart, removeFromCart }}>
            <ItemCard />
        </CartContext.Provider>
        
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <CartContext.Provider value={{ isCartOpen, setIsCartOpen, cart, addToCart, removeFromCart }}>
            <ItemCard />
        </CartContext.Provider>
    );
	expect(asFragment()).toMatchSnapshot();
});