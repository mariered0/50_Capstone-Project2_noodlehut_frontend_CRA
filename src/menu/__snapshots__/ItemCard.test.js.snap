// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
    >
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        src="image_unavailable.png"
      />
      <div
        class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        />
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Add
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
